lib/plugin.js

```diff
@@ -226,28 +232,58 @@ internals.implementation = function (server, options) {
 };
 
 
+const getTokenFromCookie = (cookieHeader, cookieName) => {
+
+    if (!(cookieHeader || cookieName)) {
+        return null;
+    }
+
+    const cookies = cookieHeader.split(/;(\s+)?/);
+
+    // find the cookie with the cookie name given in the settings
+    for (let i = 0; i < cookies.length; i += 1) {
+        const [name, value] = cookies[i].split('=');
+
+        if (name === cookieName) {
+            return value;
+        }
+    }
+
+    return '';
+};
+
+
 internals.token = function (request, settings, missing, unauthorized) {
 
-    const authorization = request.headers.authorization;
+    // if both the headerName and cookieName are given false then we should fallback to
+    // authorization header as the default
+    const authorization =
+        request.headers[settings.headerName]
+            || getTokenFromCookie(request.headers.cookie, settings.cookieName)
+            || request.headers.authorization;
```

Comment:
This fallback behaviour does not seem safe, and could introduce possibly security issues. I would much rather that the settings uniquely specify where the authorization is to be extracted from.

Reply:
Ok I agree with that. But if the last fallback thing `request.headers.authorization` is removed then only one of them will be used. If headerName is not set to false then header will be used, if the user wanted to use cookie then they need to put headerName to false so it goes to the cookieName.

Reply:
Your updated logic is still not safe, as it still includes a fallback behaviour based on the request headers.

There should be _no fallback behaviour_. Depending on the configuration it should either use one named header, or one named cookie. If a user wants to enable multiple ways to provide this a token, they can register the jwt auth multiple times.

Reply:
Oh ok got it. You want to use the settings to determine this rather than the header directly. Sorry I didn't quite get what you were referring by fallback behaviour